src/platform/common/configMigration.ts

```diff
@@ -31,44 +31,71 @@ export class ConfigMigration {
     constructor(private readonly jupyterConfig: WorkspaceConfiguration) {}
 
     public async migrateSettings() {
+        const migratedSettings: Thenable<void>[] = [];
         for (let prop of Object.keys(ConfigMigration.migratedSettings)) {
-            await this.migrateSetting(prop, ConfigMigration.migratedSettings[prop]);
+            migratedSettings.push(...this.migrateSetting(prop, ConfigMigration.migratedSettings[prop]));
+        }
+        try {
+            await Promise.all(migratedSettings);
+        } catch (e) {
+            handleSettingMigrationFailure(e);
         }
     }
 
-    private async migrateSetting(oldSetting: string, newSetting: string) {
+    private migrateSetting(oldSetting: string, newSetting: string) {
         const oldDetails = this.jupyterConfig.inspect(oldSetting);
         const newDetails = this.jupyterConfig.inspect(newSetting);
 
-        try {
-            if (oldDetails?.workspaceValue !== undefined) {
-                if (newDetails?.workspaceValue === undefined) {
-                    await this.jupyterConfig.update(
-                        newSetting,
-                        oldDetails.workspaceValue,
-                        ConfigurationTarget.Workspace
-                    );
-                }
-                await this.jupyterConfig.update(oldSetting, undefined, ConfigurationTarget.Workspace);
+        const migratedSettings: Thenable<void>[] = [];
+        if (oldDetails?.workspaceValue !== undefined) {
+            let promise: Thenable<void> = Promise.resolve();
+            if (newDetails?.workspaceValue === undefined) {
+                promise = this.jupyterConfig.update(
+                    newSetting,
+                    oldDetails.workspaceValue,
+                    ConfigurationTarget.Workspace
+                );
```

Comment:
```ts
            const promise: Thenable<void> = newDetails?.workspaceValue === undefined
                ? this.jupyterConfig.update(
                    newSetting,
                    oldDetails.workspaceValue,
                    ConfigurationTarget.Workspace)
                : Promise.resolve();
```

could use a ternary for these since it's a simple assignment but that's a nit and the if works.